File: server/utils/lg/orderStatus.go
package lg

import "github.com/flipped-aurora/gin-vue-admin/server/model/lg"

func OrderStatus(order lg.Order) (status string) {
	if order.LogoutID != nil {
		status = "销函"
	} else if order.ClaimID != nil && *order.Claim.AuditStatus == 2 {
		status = "理赔"
	} else if order.RefundID != nil && *order.Refund.AuditStatus == 2 {
		status = "退函"
	} else if order.DelayID != nil && *order.Delay.AuditStatus == 2 {
		status = "延期"
	} else if order.LetterID != nil {
		status = "已开"
	} else {
		status = "未开"
	}
	return
}

func AuditStatus(statusInt int64) (status string) {
	switch statusInt {
	case 1:
		status = "待审"
	case 2:
		status = "通过"
	case 3:
		status = "拒绝"
	default:
		status = ""
	}
	return
}

File: web/src/utils/lg/orderStatus.js
export const orderStatus = (order) => {
  if (order.revoke != null) {
    return '已撤'
  } else if (order.logout != null) {
    return '销函'
  } else if (order.claim != null && order.claim.auditStatus === 2) {
    return '理赔'
  } else if (order.refund != null && order.refund.auditStatus === 2) {
    return '退函'
  } else if (order.delay != null && order.delay.auditStatus === 2) {
    return '延期'
  } else if (order.letter != null) {
    return '已开'
  } else {
    return '未开'
  }
}

export const orderStatusType = (order) => {
  if (order.revoke != null) {
    return 'warning'
  } else if (order.logout != null) {
    return 'info'
  } else if (order.claim != null && order.claim.auditStatus === 2) {
    return 'danger'
  } else if (order.refund != null && order.refund.auditStatus === 2) {
    return 'info'
  } else if (order.delay != null && order.delay.auditStatus === 2) {
    return ''
  } else if (order.letter != null) {
    return 'success'
  } else {
    return 'warning'
  }
}
